enclave_build/src/lib.rs

```diff
@@ -266,43 +275,62 @@ impl<'a> Docker2Eif<'a> {
             .output()
             .map_err(|e| EnclaveBuildError::LinuxKitError(format!("{:?}", e)))?;
         if !output.status.success() {
-            eprintln!(
-                "Linuxkit reported an error while creating the bootstrap ramfs: {:?}",
-                String::from_utf8_lossy(&output.stderr)
-            );
             return Err(EnclaveBuildError::LinuxKitError(format!(
-                "{:?}",
+                "Linuxkit reported an error while creating the bootstrap ramfs: {:?}",
                 String::from_utf8_lossy(&output.stderr)
             )));
         }
 
-        // Prefix the docker image filesystem, as expected by init
-        let output = Command::new(&self.linuxkit_path)
-            .args([
-                "build",
-                "-docker",
-                "-name",
-                &customer_ramfs,
-                "-format",
-                "kernel+initrd",
-                "-prefix",
-                "rootfs/",
-                ramfs_with_rootfs_config_file.path().to_str().unwrap(),
-            ])
-            .output()
-            .map_err(|e| EnclaveBuildError::LinuxKitError(format!("{:?}", e)))?;
-        if !output.status.success() {
-            eprintln!(
-                "Linuxkit reported an error while creating the customer ramfs: {:?}",
-                String::from_utf8_lossy(&output.stderr)
-            );
-            return Err(EnclaveBuildError::LinuxKitError(format!(
-                "{:?}",
-                String::from_utf8_lossy(&output.stderr)
-            )));
+        // If the docker daemon should be used, then call linuxkit with the '-docker' flag, which
+        // makes linuxkit search the image in the docker cache first.
+        // Otherwise, do not add the flag and let it pull the images itself, without docker.
+        if self.image_manager.use_docker() {
```

Comment:
At this moment we only initialize it as `DockerImageManager`. What's the point for this if here?
And if you plan to also initialize `OciImageManager` in this module then can not we store this `isDocker` externally rather providing a method from the inside?

Reply:
Removing this function and replacing it with a variable or a type match.